### DISCUSSION TITLE
Blog #6

### DISCUSSION THREAD

**Discussion by AlinaBoess** [AlinaBoess/SoftwareEngineeringProjektTINF23B5#6] [General]
# Sechster Blogeintrag - Fortschritte bei der zweiten Funktion

Schön, dass ihr es wieder zu uns geschafft habt! 😊 Diese Woche stand bei uns die Entwicklung einer weiteren Kernfunktion im Fokus, auch wenn es aufgrund von Krankheiten und anderen Verpflichtungen nicht ganz wie geplant voranging. Trotzdem möchten wir euch ein Update zu unserem Fortschritt geben.

## Unser Ziel: Login-Funktion für User

Nach der erfolgreichen Anzeige der Tische in der letzten Woche haben wir uns dieses Mal auf die Implementierung einer **Login-Funktion** konzentriert. Der Login ist ein essenzieller Bestandteil unseres Systems und ermöglicht den Benutzern den Zugang zu personalisierten Funktionen.

### Fortschritte

- **Frontend (abgeschlossen):**  
  Die Benutzeroberfläche der Login-Seite ist vollständig fertiggestellt und bietet ein klares, intuitives Design. Hier ist ein kleiner Einblick, wie der Login-Bereich aussieht:  
  ![grafik](https://github.com/user-attachments/assets/4a3b89ff-0f57-42fa-91ff-fcb7306a4d71)

- **Backend (in Arbeit):**  
  Die serverseitige Verarbeitung der Login-Daten sowie die Authentifizierung stehen kurz vor der Fertigstellung. Momentan arbeiten wir noch an der sicheren Anbindung unserer Datenbank, um die Benutzeranmeldung reibungslos zu ermöglichen.  

### Zusätzliche Fortschritte: Datenbankstruktur überarbeitet

Diese Woche haben wir unsere **Datenbankstruktur** einer gründlichen Überarbeitung unterzogen. Ziel war es, sie besser an die Anforderungen der neuen Funktionen anzupassen. Hier seht ihr ein Diagramm unserer aktuellen Datenbankstruktur:  
![grafik](https://github.com/user-attachments/assets/f3ab6152-a1a0-4960-a8b2-f8123bd1e66c)  

Hierzu haben wir folgende Enums definiert:
```
Enum role {
   CUSTOMER
   ADMIN
}
Enum actionType {
   ADD
   MODIFIY
   DELETE
}
```
Die neue Struktur hilft uns, die Benutzer- und Reservierungsdaten effizient zu verwalten und zukünftige Erweiterungen einfacher umzusetzen.

### Herausforderungen und Ausblick

Diese Woche war für unser Team nicht einfach:  
- **Zeitprobleme:** Da einige von uns krank waren, mussten wir die Arbeit neu verteilen.  
- **Backend-Integration:** Das Zusammenspiel zwischen Datenbank, Backend und Frontend erwies sich als anspruchsvoller als erwartet.  

Dennoch sind wir optimistisch, dass wir die Login-Funktion nächste Woche vollständig abschließen können. 

## Sonstige Aufgaben

Neben der Arbeit an der Login-Funktion haben wir uns noch mit Folgendem beschäftigt:  
- Weitere Strukturierung unseres GitHub-Repositorys zur besseren Übersicht.  
- Planung zukünftiger Funktionen wie der Benutzerrollenverwaltung.  

Wir hoffen, euch in unserem nächsten Blogeintrag die fertige Login-Funktion präsentieren zu können! Schaut gerne wieder vorbei und lasst uns wissen, was ihr denkt. 😊  

Liebe Grüße  
Lukas, Alex, Moumen, Yahya und Alina

**Comment by Mimikry2** [comment]
Moin,

Die Benutzeroberfläche der Login-Seite ist übersichtlich und funktional gestaltet. Sie ermöglicht es Nutzern, sich mithilfe ihrer E-Mail-Adresse und ihres Passworts einfach und intuitiv anzumelden. Ich vermute mal der Sign Up Button fehlt noch? Und wenn die beiden Eingabefelder zusammen anfangen und enden, also damit das eine nicht so raussteht, sieht das vielleicht etwas schöner aus.

Viele Grüße,
Tim, Dennis, Nico

**Comment by LeonRother** [comment]
Hey,
Sieht schon ziemlich gut aus, besonders interessant finde ich eure Überarbeitung der Datenbankstruktur und die Einführung von Enums – das macht euer System sicher flexibler für zukünftige Erweiterungen. Was wir uns noch gefragt haben ist: Ob ihr für die sichere Anbindung der Datenbank und die Authentifizierung spezielle Sicherheitsmaßnahmen wie Hashing von Passwörtern oder die Verwendung von Token (z. B. JWT) eingeplant habt?
Grüße Team SE-Platzverwaltung

**Comment by SafaeKa** [comment]
Hi,

Wir hoffen, dass alle in eurem Team wieder gesund sind :) Die Wochenaufgabe habt ihr trotzdem toll geschafft!

Das UI erfüllt die geplante Funktionalität und euer aktualisiertes Klassendiagramm ist sehr übersichtlich. Welche Software habt ihr dafür verwendet? Ich gehe davon aus, dass sich der Benutzer beim ersten Einloggen registrieren muss, oder? Wenn das der Fall ist, habt ihr es vor, eine Anmeldeoption zu integrieren?

GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas

**Comment by GermanJesus-lul** [comment]
Hey,

Eure Arbeit dieser Woche sieht schonmal sehr vielversprechend aus. Außer dem von anderen angesprochenen (möglicherweise) fehlenden "SignUp"-Button und dem rudimentären Design, habe ich nichts am Frontend anzumerken.
Mich würde aber wie ihr mit Passwörtern, Autorisierung und Authentifizierung umgeht.
Ich finde es gut, dass ihr eure Datenbank ausführlich plant. Wäre es aber vielleicht sinnvoll die "role" eines Users in einer separaten Tabelle zu speichern? In dieser dann auch die Zugriffsrechte hinterlegt sind? Dabei denke ich vor allem an Restaurantbesitzer, die ja nur Reservierungen/Tische für ihr eigenes Restaurant verwalten sollten können.

Mit freundlichen Grüßen,
Julius und das Rezeptbuch Team

**Comment by stefanschfr** [comment]
Hallo,

eure Fortschritte sind beeindruckend, besonders die überarbeitete Datenbankstruktur mit den Enums! Das sorgt sicher für eine bessere Erweiterbarkeit. Beim Frontend fällt mir auf, dass der Login-Bereich funktional und gut gestaltet ist. Ergänzend dazu: Plant ihr, den "Sign Up"-Button direkt mit einer Registrierungsmöglichkeit zu verknüpfen? Das würde die User Experience noch runder machen.

Außerdem wäre es optisch ansprechender, wenn die Eingabefelder für E-Mail und Passwort gleich anfangen und enden – das sorgt für ein harmonischeres Layout.

Weiter so, wir sind gespannt auf die nächste Woche! 😊

Viele Grüße
